lib/jna/clink_jna.cc

```diff
@@ -106,21 +117,25 @@ void SparseVector_delete(SparseVectorJNA *vector) { delete vector; }
 
 SparseVectorJNA *OneHotEncoderModel_transform(clink::OneHotEncoderModel *model,
                                               const int value,
-                                              const int columnIndex) {
-  auto sparse_vector = model->transform(value, columnIndex);
-  CLINK_JNA_HANDLE_ERROR(sparse_vector.takeError())
+                                              const int column_index) {
+  auto sparse_vector = model->transform(value, column_index);
+  CLINK_JNA_HANDLE_ASYNC_ERROR(sparse_vector)
+  if (sparse_vector.IsError()) {
```

Comment:
Should we wait for the returned AsyncValueRef to be available before checking whether it incurs error?

And it seems that the functionality of `CLINK_JNA_HANDLE_ASYNC_ERROR` and `if (sparse_vector.IsError())` overlaps with each other. Could you check whether we can simplify the logic here?

We also need to add Java doc for `CLINK_JNA_HANDLE_ASYNC_ERROR` if we decide to add it.

Reply:
I'll add `Await()` and documents to this macro. 

Simplifying the logic by merging `CLINK_JNA_HANDLE_ASYNC_ERROR` and `if (sparse_vector.IsError())` might not be simple, as `CLINK_JNA_HANDLE_ASYNC_ERROR` does not know whether its caller needs to return NULL, -1 or nothing. I'll leave a TODO here first.